volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -1224,6 +1236,221 @@ def get_inode(self):
+            null_mac_addr_bytes = b"\x00" * self.addr_len
+            null_mac_addr = self._format_as_mac_address(null_mac_addr_bytes)
+            mac_addr = self._format_as_mac_address(self.perm_addr)
+            if mac_addr != null_mac_addr:
+                return mac_addr
+
+        parent_layer = self._context.layers[self.vol.layer_name]
+        try:
+            hwaddr = parent_layer.read(self.dev_addr, self.addr_len, pad=True)
+        except exceptions.InvalidAddressException:
+            vollog.debug(
+                f"Unable to read network inteface mac address from {self.dev_addr:#x}"
+            )
+            return None
+
+        return self._format_as_mac_address(hwaddr)
+
+    def _get_flag_choices(self) -> Dict:
+        """Return the net_device flags as a list of strings"""
+        vmlinux = linux.LinuxUtilities.get_module_from_volobj_type(self._context, self)
+        try:
+            # kernels >= 3.15
+            net_device_flags_enum = vmlinux.get_enumeration("net_device_flags")
+            choices = net_device_flags_enum.choices
+        except exceptions.SymbolError:
+            # kernels < 3.15
+            choices = NET_DEVICE_FLAGS
+
+        return choices
+
+    def _get_net_device_flag_value(self, name):
+        """Return the net_deivce flag value based on the flag name"""
+        return self._get_flag_choices()[name]
+
+    def get_flag_names(self) -> List[str]:
+        """Return the net_deivce flags as a list of strings
+
+        Returns:
+            List[str]: A list of flag names
+        """
+        return list(self._get_flag_choices())
+
+    @property
+    def promisc(self):
+        """Return if this network interface is in promiscuous mode.
+
+        Returns:
+            bool: True if this network interface is in promiscuous mode. Otherwise, False
+        """
+        return self.flags & self._get_net_device_flag_value("IFF_PROMISC") != 0
+
+    def get_net_namespace_id(self) -> int:
+        """Return the network namespace id for this network interface.
+
+        Returns:
+            int: the network namespace id for this network interface
+        """
+        nd_net = self.nd_net
+        if nd_net.has_member("net"):
+            # In kernel 4.1.52 the 'nd_net' member type was changed from
+            # 'struct net*' to 'possible_net_t' which has a 'struct net *net' member.
+            net_ns_id = nd_net.net.get_inode()
+        else:
+            # In kernels < 4.1.52 the 'nd_net'member type was  'struct net*'
+            net_ns_id = nd_net.get_inode()
+
+        return net_ns_id
+
+    def get_operational_state(self) -> str:
+        """Return the netwok device oprational state (RFC 2863) string
+
+        Returns:
+            str: A string with the operational state
+        """
+        if self.operstate >= len(IF_OPER_STATES):
+            vollog.warning(f"Invalid net_device operational state '{self.operstate}'")
+            return "INVALID"
```

Comment:
Rather than "INVALID" as a string would a UnparsableValue from volatility3/volatility3/framework/renderers be better here?